static/js/protein/chart.ts

```diff
@@ -21,18 +21,80 @@ const XLINKNS = "http://www.w3.org/1999/xlink";
 
 const HEIGHT = 224;
 const WIDTH = 500;
+const MARGIN = { top: 70, right: 50, bottom: 10, left: 50 };
 
+// interface for protein page datatypes which return number values
+export interface ProteinPropDataNumType {
+  name: string;
+  value: number;
+}
+
+// interface for protein page datatypes which return string values
+export interface ProteinPropDataStrType {
+  name: string;
+  value: string;
+}
 /**
  * Draw bar chart for tissue score.
  */
 export function drawTissueScoreChart(
   id: string,
   data: { name: string; value: string }[]
 ): void {
-  // TODO: convert data to tmpData format.
-  const tmpData = [{ x1: 20, x2: 60, y1: 30, y2: 50 }];
+  /*
+  Convert the tissue expression level to numerical values 
+  * @param val value of the data array 
+  */
+  function convertData(val) {
+    if (val == "ProteinExpressionNotDetected") {
+      return 0;
+    } else if (val == "ProteinExpressionLow") {
+      return 1;
+    } else if (val == "ProteinExpressionMedium") {
+      return 2;
+    } else if (val == "ProteinExpressionHigh") {
+      return 3;
+    }
+  }
+
+  const reformattedData = data.map((item) => {
+    const newObj = {} as ProteinPropDataNumType;
```

Comment:
nit: lines 61-65 can be replaced with:

```
return {
    name: item.name,
    value: convertData(item.value),
}
```

and to enforce type, you can specify that the type of `reformattedData` is `ProteinPropDataNumType[]`

Reply:
Done !